lib/mini_autobot/google_sheets.rb

```diff
@@ -0,0 +1,77 @@
+module MiniAutobot
+
+  # Class that supports integration with Google Sheets
+  # See documentation here: https://developers.google.com/drive/v3/web/about-auth to set up OAuth 2.0
+  # access to your sheets. Place the info in google_drive_config.json in your config/mini_autobot folder
+  #
+  # To use, use the -g or --google_sheets parameter with the id of your sheet
+  # Example: -g 1gWbpPq7rrTdNtSJSN2ljDlPE-okf0w3w7ai5PlhY8bk
+  class GoogleSheets
+    require 'rubygems'
+    require 'google/api_client'
+    require 'google_drive'
+    require 'json'
+
+    def initialize
+      @session = session
+      @spreadsheet = spreadsheet
+      @worksheet = worksheet
+      @automation_serial_column = automation_serial_column
+      @selected_browser_column = selected_browser_column
+    end
+
+    def session
+      GoogleDrive.saved_session(MiniAutobot.root.join('config/mini_autobot', 'google_drive_config.json'))
+    end
+
+    def spreadsheet
+      @session.spreadsheet_by_key(MiniAutobot.settings.google_sheet)
```

Comment:
My thinking behind this was to set it up in such a way that the interface would be easy to use - everything is built off of whether you pass in the -g parameter with a sheet ID - if you don't have that, the GoogleSheets class never gets accessed - if you do have it, then the GoogleSheets class is activated and works as intended because the sheet is stored in settings - I think I will make all of these methods private to ensure that they are used as intended.